src/test/java/org/pesmypetcare/webservice/service/MedicationServiceTest.java

```diff
@@ -0,0 +1,159 @@
+package org.pesmypetcare.webservice.service;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.pesmypetcare.webservice.dao.MedicationDao;
+import org.pesmypetcare.webservice.entity.MedicationEntity;
+import org.pesmypetcare.webservice.error.DatabaseAccessException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class MedicationServiceTest {
+    private static List<Map<List<String>, Object>> medicationList;
+    private static MedicationEntity medicationEntity;
+    private static String owner;
+    private static String petName;
+    private static String dateName;
+    private static String date;
+    private static String date2;
+    private static String field;
+    private static int value;
+    private static String GET_EXCEPTION_MSG = "Should return an exception when retrieving a Medication fails";
+
+    @Mock
+    private MedicationDao medicationDao;
+
+    @InjectMocks
+    private MedicationService service = new MedicationServiceImp();
+
+    @BeforeAll
+    public static void setUp() {
+        medicationList = new ArrayList<>();
+        medicationEntity = new MedicationEntity();
+        dateName = "2020-02-13/Cloroform";
+        date = "2020-02-13";
+        date2 = "2021-02-13";
+        owner = "Pepe05";
+        petName = "Camper";
+        field = "quantity";
+        value = 3;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['3' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)